index.js

```diff
@@ -1,95 +1,131 @@
 // ***** Question 1 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 1", "color: red") 
+console.log("%cQuestion 1", "color: red") 
 
-// console.log(counter) 
+let counter = 1
+
+counter = 2
+console.log(counter) 
 // // => 2
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 // ***** Question 2 *****
 
 // *** Uncomment the lines below to test
 // *** after testing, comment the line causing the error back in 
 // *** otherwise, the error will stop the rest of your code from running
-// console.log("%cQuestion 2", "color: red") 
+console.log("%cQuestion 2", "color: red") 
 
-// console.log(name)
+const name = "Patricia"
+console.log(name)
 // // => "Ian"
-// name = "Not Ian" 
+//name = "Not Patricia" 
 // // => TypeError
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 3 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 3", "color: red") 
+console.log("%cQuestion 3", "color: red") 
+
+function drinkWater(currentThirstLevel){
+  console.log("Man I sure am thirsty")
+  currentThirstLevel -= 1
+  console.log("Ahh that hits the spot")
+  return currentThirstLevel
+}
 
-// console.log(drinkWater(12))
+console.log(drinkWater(12))
 // // => "Man I sure am thirsty"
 // // => "Ahh that hits the spot"
 // // => 11
-// console.log("%c----------", "color: red") 
+console.log("%c----------", "color: red") 
 
 
 
 // ***** Question 4 *****
 
 // *** Uncomment the lines below to test
-// console.log("%cQuestion 4", "color: red") 
+console.log("%cQuestion 4", "color: red") 
+
+function sameSameButDifferent(num, maybeNum){
+  if (num === maybeNum){
+    console.log("same same")
+  } else if (parseInt(num) === parseInt(maybeNum)){
+    console.log("same same (but different)")
+  } else if (parseInt(num) !== parseInt(maybeNum)){
+    console.log("different")
+  }
+}
```

Comment:
This will work, but just a couple notes:

- the last `else if` can just be replaced with an `else` 
- we also don't need to use `parseInt` with `num` since we're always passing a number as an argument (parseInt is useful if we're dealing with strings)

```js
  if (num === maybeNum){
    console.log("same same")
  } else if (num === parseInt(maybeNum)){
    console.log("same same (but different)")
  } else {
    console.log("different")
  }
```